packages/salesforcedx-vscode-core/package.json

```diff
@@ -50,7 +50,7 @@
     "mock-spawn": "0.2.6",
     "nyc": "^11.0.2",
     "sinon": "^2.3.6",
-    "typescript": "2.6.2",
+    "typescript": "2.9.1",
```

Comment:
@allileong - If we are upgrading TypeScript, we should do it consistently across all modules. Usually when I do that, I do it in a separate commit to let the test run fully.

Is there a reason why we only update TypeScript for this one module?

Reply:
@vazexqi I plan to complete that as I finish this work. I just wanted to open a WIP review to make sure that I'm on the right track with my approach.

Reply:
@vazexqi that's a good idea. When I finalize my upgrade, I will do it in a separate commit